src/webapi/date.rs

```diff
@@ -1,25 +1,1166 @@
+    /// older implementations, which ignore the locales and options arguments, the locale used and
+    /// the form of the string returned are entirely implementation dependent.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString)
+    #[inline]
+    fn to_locale_string( &self, _locales: Option< &str >, _options: Option< &HashMap< &str, &str > >) -> String {
+        match _locales {
+            Some(_locales) => match _options {
+                Some(_options) => return js! (
+                    var _options = @{_options};
+
+                    return @{self.as_ref()}.toLocaleString(@{_locales}, _options);
+                ).try_into().unwrap(),
+
+                None => return js! (
+                    return @{self.as_ref()}.toLocaleString(@{_locales});
+                ).try_into().unwrap()
+            },
+
+            None => return js! ( return @{self.as_ref()}.toLocaleString(); ).try_into().unwrap()
+        }
+    }
+
+    /// The to_locale_time_string() method returns a string with a language sensitive representation
+    /// of the time portion of this date. The new locales and options arguments let applications
+    /// specify the language whose formatting conventions should be used and customize the behavior
+    /// of the function. In older implementations, which ignore the locales and options arguments,
+    /// the locale used and the form of the string returned are entirely implementation dependent.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString)
+    #[inline]
+    fn to_locale_time_string( &self, _locales: Option< &str >, _options: Option< &HashMap< &str, &str > > ) -> String {
+        match _locales {
+            Some(_locales) => match _options {
+                Some(_options) => return js! (
+                    var _options = @{_options};
+
+                    return @{self.as_ref()}.toLocaleTimeString(@{_locales}, _options);
+                ).try_into().unwrap(),
+
+                None => return js! (
+                    return @{self.as_ref()}.toLocaleTimeString(@{_locales});
+                ).try_into().unwrap()
+            },
+
+            None => return js! ( return @{self.as_ref()}.toLocaleTimeString(); ).try_into().unwrap()
+        }
+    }
+
+    // NON-STANDARD
+    // fn to_source( &self ) -> i32 { }
+
+    /// The to_s() method returns a string representing the specified Date object.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toString)
+    #[inline]
+    // naming this to to_s 1. to avoid collision with standard to_string(), and 2. to avoid linting
+    // error on toString()...
+    fn to_s( &self ) -> String {
+        js! ( return @{self.as_ref()}.toString(); ).try_into().unwrap()
+    }
```

Comment:
Again, I don't think this method should be here. (: As a Rubyist seeing `to_s` warms my heart, but if we want to expose it then we should do it as `impl From` to be idiomatic Rust (or not do it at all for now)

Reply:
🎉  for the Ruby shoutout  ! On thinking over it a little more, I have to agree. For one, it doesn't seem like this would really be that useful in practice, and two, we would need to put a little more thought into a true "to_string" method from the Rust side, in case we wanted to show what Rust thinks of the object as a whole (to be used for debugging/inspection etc.)